docs/mkdocs/documentation/day1_limited_option.md

```diff
@@ -0,0 +1,58 @@
+   It means that the in-tree module is not referenced by any other kernel modules
+3. in order for day1 functionlity to work, the node must have an functional network interface, meaning: an in-tree kernel driver for that imterface.
+   The OOT kernel module can be network driver that will replace a functional network driver.
+
+## Day1 OOT kernel module loading flow
+
+The loading of OOT kernel module utilises support of MCO.The following is the sequence of the flow:
+1. Customer applies an MCO yaml to the existing running cluster. In order to identify the necessary nodes that need to be
+   update, customer must identify/create an appropriate MachineConfigPool
+2. MCO applies reboots node by node. On any rebooted node, 2 new systemd services are deployed: pull service and load service
+3. Load service is configured to run prior to NetworkConfiguration service. It tries to pull a predefined kernel module image
+   abd then, using that image, to unload an in-tree module and load an OOT kernel module. 
+4. Pull service is configured to run after NetworkManager service. It checks if the pre-configured kernel module image is located
+   on the node file-system. If it is - then the service exists normally, and server continues with linux initialization process.
+   If not - it pull the image onto the node, and reboots the node afterwards
+
+## Kernel Module Image
+Day1 functionality uses the same DTK based image, as the one used by usual KMM Day2 flow. OOT kenrel module should be located under /opt
+base directory
+
+## In-tree module replacement
+Day1 functionality always tries to replace the in-tree kernel module with OOT kernel module. In case in-tree is not loaded, it does not affect
+the flow, the service will just continue to load the OOT kernel module.  
+
+## MCO yaml creation
+KMM provides an API to create an MCO yaml need for day1 functionality:
+```go
+ProduceMachineConfig(machineConfigName, machineConfigPoolRef, kernelModuleImage, kernelModuleName string) (string, error)
+```
+The returned output is a string representation of an MCO yaml to be applied. It is up to the customer to apply this yaml.
+The parameters are:
+1. _machineConfigName_ - The name of the MCO yaml. It will be set as a name parameter of the metadata of MCO yaml
+2. _machineConfigPoolRef_ - the machineConfigPool name that will be used in order to identify the targeted nodes
+3. _kernelModuleImage_ - container image that includes the OOT kernel module.
+4. _kernelModuleName_ - the name of the OOT kernel module. This parameter will be used both to unload the in-tree kernel module
+   ( if loaded into the kernel) and to load the OOT kernel module
+
+The API is located under _pkg/mcproducer_ package of hte KMM source code. There is no need to KMM operator to be running in order for the customer
+to use Day1 functionality. Customer only need to import _pkg/mcproducer_ package into it's operator/utility code, call the API and to apply the produced
+MCO yaml onto the cluster
```

Comment:
```suggestion

The returned output is a string representation of the MCO YAML manifest to be applied.
It is up to the customer to apply this YAML.

The parameters are:

- `machineConfigName`: the name of the MCO YAML manifest. It will be set as the `name` parameter of the metadata of MCO YAML manifest.
- `machineConfigPoolRef`: the `MachineConfigPool` name that will be used in order to identify the targeted nodes
- `kernelModuleImage`: the name of the container image that includes the OOT kernel module.
- `kernelModuleName`: the name of the OOT kernel module. This parameter will be used both to unload the in-tree kernel module
   (if loaded into the kernel) and to load the OOT kernel module.

The API is located under `pkg/mcproducer` package of hte KMM source code.
There is no need to KMM operator to be running in order for the customer to use Day1 functionality.
Users only need to import the `pkg/mcproducer` package into their operator/utility code, call the API and to apply the produced
MCO YAML to the cluster.
```

Reply:
Please also remove the trailing lines.